aten/src/ATen/ThreadLocalDebugInfo.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <c10/macros/Export.h>
+
+#include <memory>
+#include <string>
+
+namespace at {
+
+class CAFFE2_API ThreadLocalDebugInfoBase {
+ public:
+  ThreadLocalDebugInfoBase() {}
+  virtual ~ThreadLocalDebugInfoBase() {}
+
+  virtual std::string getString(const char* name) = 0;
```

Comment:
we could also just do dynamic_cast to whatever type needed, but it works too

Reply:
Why do we require implementing those two methods in particular? What's so special about them?